llvm/lib/SYCLLowerIR/ModuleSplitter.cpp

```diff
@@ -798,6 +799,23 @@ void ModuleDesc::dump() const {
 }
 #endif // NDEBUG
 
+void ModuleDesc::saveSplitInformationAsMetadata() {
+  // Add metadata to the module so we can identify what kind of SYCL/ESIMD split
+  // later.
+  auto *SplitMD = M->getOrInsertNamedMetadata(SYCL_ESIMD_SPLIT_MD_NAME);
```

Comment:
A nit:
If we are planning to add more metadata like this in the future, may be it will help to add a helper function?
Not a blocker though

Thanks

Reply:
I don't expect it to grow that much, the things we check in module properties is limited and doesn't grow that often. If it does blow up we can just move each MD to a function, sure, thanks.